mautic/mautic
# Add Mautic 3 upgrade script

Provides a new upgrade script to allow for upgrading M2 to M3. See https://github.com/mautic/mautic/issues/8819 for details

![image](https://user-images.githubusercontent.com/17739158/99320978-12fc4100-286d-11eb-8791-5a55ce876232.png)

**Steps to test**
Mautic 2.16.3 will add a separate upgrade script for the upgrade to Mautic 3. You will first install the Mautic 2.16.3 beta and then upgrade to Mautic 3.

1. You have three options to test this PR:
**Clone this PR**: clone this PR and go to step 3 (assuming that you have a working 2.x installation already. If you clone in an empty folder, then also follow step 2).
**Clean installation**: Download the custom [2.16.3-beta.zip package](https://github.com/dennisameling/mautic/releases/download/2.16.3-beta/2.16.3-beta.zip)
**Upgrade existing 2.x installation**: Download the custom [2.16.3-beta-update.zip package](https://github.com/dennisameling/mautic/releases/download/2.16.3-beta/2.16.3-beta-update.zip)

2. Install 2.16.3-beta (clean install or run `app/console mautic:update:apply --update-package=2.16.3-beta-update.zip` to upgrade an existing installation)

3. You have two options for testing the upgrade to M3 (please do both if you have time):
**UI upgrade**: go to your Mautic dashboard, you should see a reguler update notification like you're used to. If not, run `app/console mautic:update:find` to force search for new updates.
**CLI upgrade**: run `app/console mautic:update:find` and if the update is found, run `php upgrade_V3.php`
4. You should now be able to start the upgrade. Please share your experience with the upgrade script here!

**Testing with DDEV**
For everyone who's using DDEV, you can take these steps to test:

- Open a terminal
- `mkdir mautic3upgrade`
- `cd mautic3upgrade`
- Run `wget https://github.com/dennisameling/mautic/releases/download/2.16.3-beta/2.16.3-beta.zip`
- Run `unzip 2.16.3-beta.zip`
- Run `ddev config --webserver-type apache-fpm --php-version 7.3`
- Run `ddev start`, and you should be good to go :rocket:
